src/net.md

```diff
@@ -876,6 +877,106 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-check-broken-links]: #ex-check-broken-links
+<a name="ex-check-broken-links"/>
+## Check a webpage for broken links
+
+[![reqwest-badge]][reqwest] [![select-badge]][select] [![url-badge]][url] [![cat-net-badge]][cat-net] |
+
+Use [`Url::parse`] to create a [`Url`] and then create the base URL from that. Use [`reqwest::get`] to perform a HTTP GET request and then use [`Document::from_read`] to parse the response into a HTML document.
+
+We can then check if the document has a "base" tag. If so, we get the "href" [`attr`] from the first occurrence of the "base" tag. This is then used as the base URL.
+
+We can then retrieve all the links from the document by using [`find`] with the criteria of the [`Name`] being "a". 
```

Comment:
We already have a [extract all links from a webpage](https://rust-lang-nursery.github.io/rust-cookbook/net.html#extract-all-links-from-a-webpage-html) example, where nearly all of these information is presented. Lets build on it and not duplicate the information. This is a more advanced example and I would suggest to limit the description to only the new information not covered by earlier examples.

If you think that some of this information is not covered so far please consider updating the earlier example descriptions :+1: